cmd/kube-apiserver/app/server.go

```diff
@@ -246,56 +235,32 @@ func Run(s *options.APIServer) error {
 		glog.Fatalf("Invalid Authentication Config: %v", err)
 	}
 
-	authorizationModeNames := strings.Split(s.AuthorizationMode, ",")
-
-	modeEnabled := func(mode string) bool {
-		for _, m := range authorizationModeNames {
-			if m == mode {
-				return true
-			}
-		}
-		return false
+	privilegedLoopbackToken := uuid.NewRandom().String()
+	selfClientConfig, err := s.NewSelfClientConfig(privilegedLoopbackToken)
+	if err != nil {
+		glog.Fatalf("Failed to create clientset: %v", err)
 	}
+	client, err := s.NewSelfClient(privilegedLoopbackToken)
+	if err != nil {
+		glog.Errorf("Failed to create clientset: %v", err)
+	}
+	sharedInformers := informers.NewSharedInformerFactory(client, 10*time.Minute)
 
 	authorizationConfig := authorizer.AuthorizationConfig{
 		PolicyFile:                  s.AuthorizationPolicyFile,
 		WebhookConfigFile:           s.AuthorizationWebhookConfigFile,
 		WebhookCacheAuthorizedTTL:   s.AuthorizationWebhookCacheAuthorizedTTL,
 		WebhookCacheUnauthorizedTTL: s.AuthorizationWebhookCacheUnauthorizedTTL,
 		RBACSuperUser:               s.AuthorizationRBACSuperUser,
+		InformerFactory:             sharedInformers,
 	}
-	if modeEnabled(genericoptions.ModeRBAC) {
```

Comment:
> This check is no longer needed?

Right, the informers are only activated if they are needed and so the net result is the same.